pkg/compose/down.go

```diff
@@ -85,7 +85,8 @@ func (s *composeService) down(ctx context.Context, projectName string, options a
 
 	err = InReverseDependencyOrder(ctx, project, func(c context.Context, service string) error {
 		serviceContainers := containers.filter(isService(service))
-		err := s.removeContainers(ctx, serviceContainers, options.Timeout, options.Volumes)
+		serv := project.Services[service]
```

Comment:
In the end we use the service and not the pointer (just like in `startContainer`) if the only reason to do this is just to make sure the service is defined, might as well validate here 
``` 
serv, ok := project.Services[service] 
if !ok {
  return fmt.Error() ....
}
```

Reply:
as we run inside `InReverseDependencyOrder` service is guaranteed to exist (actually, `InReverseDependencyOrder` could pass a `*ServiceConfig`, but this could have impact on immutability)

`service` is passed as a pointer down to https://github.com/docker/compose/blob/0aacddfb1d21740e941db2881a58cfcd39c16067/pkg/compose/down.go#L304, as this same method is used to also stop orphaned container, where we don't have a defined service so need the ability to pass `nil`